CaloMC/src/CaloDigiMaker_module.cc

```diff
@@ -168,14 +176,20 @@ namespace mu2e {
       mu2e::GeomHandle<mu2e::Calorimeter> ch;
       calorimeter_ = ch.get();
 
-      ConditionsHandle<AcceleratorParams> accPar("ignored");
       ConditionsHandle<CalorimeterCalibrations> calorimeterCalibrations("ignored");
 
       if (calorimeter_->nCrystal()<1 || calorimeter_->caloInfo().getInt("nSiPMPerCrystal")<1) return;
  
-      double mbtime    = (ewMarker.spillType() == EventWindowMarker::SpillType::onspill) ?  accPar->deBuncherPeriod : ewMarker.eventLength();      
+      int waveformSize = (digitizationEnd_ - digitizationStart_) / digiSampling_;
+      if (ewMarker.spillType() != EventWindowMarker::SpillType::onspill){
+        waveformSize = (ewMarker.eventLength() - digitizationStart_) / digiSampling_;
+      }
+
+      // change digitization window to relative to event marker
+      digitizationStart_ -= timeFromProtonsToDRMarker_;
+      digitizationEnd_ -= timeFromProtonsToDRMarker_;
```

Comment:
Should this be done before calculating the waveformSize (might change the length if the spillType is Offspill)

Reply:
I think this should be fine for now, as the specifics depend on the details of how the firmware handles offspill which has probably not been implemented yet. The assumption now is that during offspill there is no jittering in the event window times and the digitization window is exactly from digitizationStart to the end of the window. This is how I've implemented it in the tracker sim for now as well.